### DISCUSSION TITLE
Error in using AggregatorV3Interface FundMe.sol in Lesson 4

### DISCUSSION THREAD

**Discussion by subhradeepsamantray** [Cyfrin/foundry-full-course-cu#635] [Q&A] [answered]
Hello All,

I am getting an Error while trying to access the functions of the AggregatorV3Interface while trying to deploy the FundMe.sol Contract.
**Error -** 
call to FundMe.getVersion errored: VM error: revert.

revert
	The transaction has been reverted to the initial state.
Note: The called function should be payable if you send value and the value you send should be less than your current balance.
Debug the transaction to get more information. 

I am Not able to figure out what is the error happening here. Please help. 
![Screenshot (92)](https://github.com/Cyfrin/foundry-full-course-f23/assets/45929186/cb21e8d4-f2da-40b5-ac30-4b6c1718585b)

**Comment by theirrationalone** [comment]
Please make `fund` function payable

Like this... 

```solidity
function fund() public payable {
    // statements... 
}
```

or please make any function payable in which you wanna send value as like `fund` function.

And Please make sure you're performing executions onto the latest **`Deployed`** FundMe Contract. Check If maybe You forgot to **`Deploy`** latest FundMe Contract.

**Reply by subhradeepsamantray** [reply]
Hello, Actually the fund function is already Payable.
The contract got deployed and working fine now. The problem was probably that I was trying to deploy the contract in the Remix VM instead of Sepolia Testnet, that's why it was showing this error.

**Reply by theirrationalone** [reply]
Sorry! I missed that point... 

But you can also deploy contracts that have any payable function on vm. 

You might forgot to fill `value` input box above deployed contracts interactions side menu...

If you, then fill an eligible value in wei Or you're on remix vm you can put huge amount of fake eths to like 1 ether Or 2 ether Or even 1000 ether. Then after click on `fund` button... Transaction will hopefully get Passed!

**Reply by theirrationalone** [reply]
Of course, you forgot see this... 
![IMG_20230815_115523.png](https://github.com/Cyfrin/foundry-full-course-f23/assets/96755510/097eed89-bcc0-4c45-b7cf-05b1dfa39b8a)

You will also need to fill that value input with an eligible amount when will try deploy your contract on real testnets i.e. sepolia, goerli, etc...

**Reply by theirrationalone** [reply]
You might filling that input box value smaller than your MINIMUM USD contstant value

**Reply by theirrationalone** [reply]
Would you like to share your PriceConverter.sol & FundMe.sol with me... 

Or if you have its repo then share that with me instead.

**Reply by subhradeepsamantray** [reply]
Actually I am in the middle of the lesson. Not started with this Price Converter part. Will complete it and share the repo

**Reply by theirrationalone** [reply]
So that's the reason actually, 

Can you share your fund function's code with me i want to check why it's getting failed.

**Reply by subhradeepsamantray** [reply]
But In Sepolia Testnet, the contract is working fine, even with the Value 0, because in my understanding we don't need an initial value in the contract , right?. 
Don't know why it is not working in Remix VM. Also in the video, Patrick is deploying the contract in the testnet, maybe there is some reason. because the Aggregator Interface maybe working on testnet only.

**Reply by theirrationalone** [reply]
Ooohhh yes yes yes you want to get aggregator's version value.. There we don't need any value... 

I got your point.... Sorry for misunderstanding your question... 


you might be importing it wrong...

**Reply by theirrationalone** [reply, marked as answer]
> But In Sepolia Testnet, the contract is working fine, even with the Value 0, because in my understanding we don't need an initial value in the contract , right?. 
> Don't know why it is not working in Remix VM. Also in the video, Patrick is deploying the contract in the testnet, maybe there is some reason. because the Aggregator Interface maybe working on testnet only. 

This is the answer of your question.... 

Vm is just a fake environment.... You need switch to a real testnet and that's exactly what you're doing... 

Here we need some **offchain** data from oracle chainlink and oracle chainlink provides data to testnets and mainnet only.... `remix vm` is just a virtual fake environment so that's why we can't demand offchain data on there.

**Reply by theirrationalone** [reply]
Sorry! for all the replies i discussed above, i was thinking you're having some trouble in converting eth/usd values.

**Reply by subhradeepsamantray** [reply]
No problem, actually your replies helped me figure out the issue . Thanks 🙂
